### PR TITLE
EWS Reporting Page: Date filter doesn't seem to work

### PR DESCRIPTION
**Pull request by agrzywinski**
http://manage.dimagi.com/default.asp?162839

### CODE REVIEW

File: custom/ewsghana/reports/__init__.py [hunk 1]
```diff
@@ -127,12 +128,17 @@ def supply_points_list(self, location_id=None):
...
         supply_points = supply_points if supply_points else all_supply_points
-        last_period_st, last_period_end = calculate_last_period(self.config['enddate'])
         return StockTransaction.objects.filter(
             case_id__in=supply_points,
-            report__date__range=[last_period_st, last_period_end]
+            report__date__range=[self.config['startdate'], self.config['enddate']]
         ).distinct('case_id').values_list('case_id', flat=True)
 
+    def datetext(self):
+        today = datetime.now().replace(hour=0, minute=0, second=0, microsecond=0)
```
Comment (czue) [inline custom/ewsghana/reports/__init__.py]: this should be `utcnow()`

File: custom/ewsghana/utils.py [hunk 2]
```diff
@@ -58,7 +58,11 @@ def make_url(report_class, domain, string_params, args):
 
 
 def calculate_last_period(enddate):
-    last_th = enddate - timedelta(days=enddate.weekday()) + timedelta(days=3, weeks=-1)
```
Comment (czue) [inline custom/ewsghana/utils.py]: it's pretty hard to understand what this function is supposed to do - maybe you could add a comment or a doc string?

File: custom/ewsghana/utils.py [hunk 3]
```diff
@@ -57,8 +57,16 @@ def make_url(report_class, domain, string_params, args):
         return None
 
 
+#Calculate last full period (Friday - Thursday)
```
Comment (dimagimon) [inline custom/ewsghana/utils.py]: E265 block comment should start with '# '

File: custom/ewsghana/utils.py [hunk 4]
```diff
@@ -57,8 +57,16 @@ def make_url(report_class, domain, string_params, args):
...
 def calculate_last_period(enddate):
-    last_th = enddate - timedelta(days=enddate.weekday()) + timedelta(days=3, weeks=-1)
+    #checking if Thursday was already in this week
```
Comment (dimagimon) [inline custom/ewsghana/utils.py]: E265 block comment should start with '# '

File: custom/ewsghana/utils.py [hunk 5]
```diff
@@ -57,8 +57,16 @@ def make_url(report_class, domain, string_params, args):
...
+    i = enddate.weekday() - 3
+    if i < 0:
+        #today is Monday, Tuesday or Wednesday -> calculate Thursday from previous week
```
Comment (dimagimon) [inline custom/ewsghana/utils.py]: E265 block comment should start with '# '

File: custom/ewsghana/utils.py [hunk 6]
```diff
@@ -57,8 +57,16 @@ def make_url(report_class, domain, string_params, args):
...
+        last_th = enddate + timedelta(days=-i, weeks=-1)
+    else:
+        #today is Thursday, Friday, Saturday or Sunday -> calculate Thursday from this week
```
Comment (dimagimon) [inline custom/ewsghana/utils.py]: E265 block comment should start with '# '

**Comment by agrzywinski** [comment]
@czue

**Comment by czue** [comment]
:+1: when tests pass

**Pull request merged**